applicationset/generators/pull_request.go

```diff
@@ -73,10 +74,10 @@ func (g *PullRequestGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 	params := make([]map[string]interface{}, 0, len(pulls))
 
 	// In order to follow the DNS label standard as defined in RFC 1123,
-	// we need to limit the 'branch' to 50 to give room to append/suffix-ing it
-	// with 13 more characters. Also, there is the need to clean it as recommended
+	// we need to limit the 'branch' to 63.
+	// Also, there is the need to clean it as recommended
 	// here https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
-	slug.MaxLength = 50
+	slug.MaxLength = 63
 
```

Comment:
I wonder, is the comment just plainly wrong? It talks about suffixing the slug with 13 additional characters. If we increase the max. length to 63 here, it can't be suffixed anymore.

Can you please elaborate on this change?

Reply:
I checked the code, i don't find the code for `generate`. So I think it is ok to change the max length of  branch-slug to 63, and it is was mentioned in https://github.com/argoproj/argo-cd/issues/10622.